x/dsmr/node_test.go

```diff
@@ -994,6 +1022,287 @@ func TestNode_NewBlock_IncludesChunkCerts(t *testing.T) {
 	}
 }
 
+// TestDuplicateChunksElimination tests that duplicate chunks that have appeared before are getting correctly eliminated.
+func TestDuplicateChunksElimination(t *testing.T) {
+	r := require.New(t)
+
+	node := newTestNode(t)
+
+	blk := Block{
+		ParentID: ids.GenerateTestID(),
+		Hght:     1,
+		Tmstmp:   1,
+		blkID:    ids.GenerateTestID(),
+	}
```

Comment:
We should be building this block through `node.BuildBlock` and calling `Verify` prior to calling `Accept` (see other tests as an example). This is technically not a valid block because it references an unknown parent instead of the last accepted block which would be caught if `Verify` was called on it. Building a block should generally take the form of calling `BuildChunk -> BuildBlock -> Verify -> Accept`